### DISCUSSION TITLE
UEFI Interactive Shell v2.2 mapping table

### DISCUSSION THREAD

**Discussion by HamzaRadwan2** [utmapp/UTM#3486] [UTM Help (macOS)]
<img width="800" alt="Screen Shot 2022-01-14 at 5 24 44 PM" src="https://user-images.githubusercontent.com/42738344/149540297-b1265fc6-d859-424d-b25e-57b0b904cdb4.png">
i have this error every time i open the virtual machine and idk how to fix it please help

im on Apple silicon

**Comment by conath** [comment]
This is not an error. It is what happens when you don't have an operating system installed in a UTM VM. I invite you to watch [my tutorial video]() to learn about how UTM works (watch from beginning until 4:49), and read the [Windows for ARM install guide](https://github.com/utmapp/UTM/wiki/Install%20Windows%20ARM64%20on%20Apple%20M1) on the wiki for more specific information.

**Reply by yazdaniit** [reply]
where is your tutorial video ?

**Reply by AngusC222** [reply]
@yazdaniit, [Here is his video](https://www.youtube.com/watch?v=KsNoGPczavw)

**Comment by Rtoax** [comment]
I got same situation.

**Reply by shareef-dweikat** [reply]
Did you solve it?

**Comment by Rtoax** [comment]
参考我的：

```
[rongtao@ft-74 images]$ efibootmgr efiboot.img 
BootCurrent: 0001
Timeout: 3 seconds
BootOrder: 0001,0002,0000,0003
Boot0000* Setup
Boot0001* CentOS SCSI Device
Boot0002* UEFI Shell
Boot0003  UEFI BootManagerMenuApp
```

**Reply by nicrob97** [reply]
I have the same issue.
I want to Emulate Windows 10 x86 architecture (and not Virtualize it) from my mac M1.
When I do the Emulation process using the iso of my version of windows 10 x86 at the launch of the VM i got this annoying UEFI Interactive Shell v2.2 and I don't know what to do.
Normally I would like to be able to install windows.. 
What should I type/do here?
Thanks guys :)

**Reply by conath** [reply]
[this video guide](https://www.youtube.com/watch?v=KsNoGPczavw) might help

**Reply by MESOLONELY** [reply]
> [this video guide](https://www.youtube.com/watch?v=KsNoGPczavw) might help

Not work on step 6

**Reply by 70599** [reply]
3:40 helped.

**Reply by garfieldnate** [reply]
@MESOLONELY I initially used the wrong version of virtio; you have to use the same 190 version that he uses in the video. I downloaded 240 and it did not work.

**Reply by CelineHarakee** [reply]
I got an error "Failed to lock byte 100" in this part... how can I fix it

**Comment by shareef-dweikat** [comment]
Any updates? I am facing this issue in my Debian VM on my MACOS Apple chip.

**Reply by conath** [reply]
Did you force-shutdown or -restart the VM using the toolbar buttons? This can corrupt the VM’s hard drive and lead to this screen. You can try booting the VM in console mode (Edit => Display) and then you can look for filesystem errors.

**Reply by debojyoti-majumder** [reply]
I am also facing the same issue.

**Comment by cpach** [comment]
FWIW I get this situation too, on a Macbook Pro with Intel CPU.

It has happened with VMs that worked fine for a while, and then this screen appears after booting.

It seems to occur after forcing shutdown of the VM. I’m guessing that the qcow2 becomes corrupted somehow?

**Reply by conath** [reply]
This is likely the problem. You can try booting the VM in console mode (Edit => Display) and then you can look for filesystem errors.

**Comment by goman64** [comment]
I resolved this issue on M1 Apple Silicon by disabling UEFI boot. Turn off the VM. Right-click and select Edit->QEMU and uncheck UEFI Boot under Tweaks.

I got the solution from this [video](https://www.youtube.com/watch?v=A3z64QUTJsM) which is about Virtualbox.

**Reply by dna246** [reply]
Thank you goman64, however, when I did that and re-ran the VM, it show as "guest has not initialized the display (yet)."
![Screen Shot 2022-09-30 at 08 27 16](https://user-images.githubusercontent.com/95900562/193170491-11097705-85a9-466c-bef7-8b1939913bee.JPG)
Please help!!

**Reply by eliazonta** [reply]
same issue :( some1 knows how to fix ?

**Reply by watermoon** [reply]
thank you very much, [goman64](https://github.com/goman64)，you save my life

**Reply by VeraxLee** [reply]
not working the utm logo even didnt show.

**Reply by DanielHallx** [reply]
very thankful!

**Reply by axelvag** [reply]
To find the button on VitualBox > 7.0, System doesn't have the button 'Enable EFI' in System BUT clic on 'Expert' settings and now the button appear !
<img width="931" alt="Capture d’écran 2025-01-02 à 13 50 35" src="https://github.com/user-attachments/assets/56426bc1-46d4-4cf3-ba22-6a2dab772b95" />

**Comment by stuartofmine** [comment]
For me, that's because I downloaded the VHDX file but boot in ISO mode.

Just create a new UTM VM and set true for  "import  VHDX". Everything is OK now.

**Reply by stuartofmine** [reply]
<img width="444" alt="截屏2022-10-13 16 13 59" src="https://user-images.githubusercontent.com/25903841/195541390-1ab343d1-76c8-4d13-884e-2bb08c335b97.png">

**Reply by abiredavid** [reply]
This worked for me

**Reply by 19KaliNoob87** [reply]
Create a new utm vm via VHDX? How do I do this

**Comment by xiaodaiweiwei** [comment]
I didn't find the setting button. How could I fix this problem?

**Comment by cOnGitHub** [comment]
Before the UEFI Interactive Shell Window appears, the following message is shown for a few seconds. 

Press any Key as long as that message is shown and Windows Installation will start. 

Of course I only noticed this solution _after_ creating some new UTM VMs as proposed in this discussion ;-)

<img width="788" alt="image" src="https://user-images.githubusercontent.com/25119948/200203290-12cf2371-3f08-43ba-876a-07972a0f9456.png">

**Reply by toddy86** [reply]
To add to this, once you have pressed any key, you will receive this screen which won't change for a good couple of minutes. Be patient and the Windows setup will eventually start. Patience on this step was my problem! I thought it was frozen and not doing anything.

![Screen Shot 2022-11-17 at 06 45 54](https://user-images.githubusercontent.com/10559757/202366690-86475157-010e-480c-8a49-c284dfa18ad1.png)

**Reply by bobmoff** [reply]
Thanks! This was the same issue as mine.

**Reply by 18474475** [reply]
牛逼

**Comment by kongmunist** [comment]
Computer: M1 Macbook Pro
Trying to run: Windows 10 x64

I fixed this with two steps
1) I need to emulate x64 on ARM64, but I instead was used to clicking "Virtualize" (only allows native CPU architecture) on UTM New OS screen instead of "Emulate" 
2) Disabling UEFI Boot / EFI Boot in the UTM options

**Reply by dt1930** [reply]
Wait, how do you disable UEFI Boot/EFI Boot in the UTM options?

**Comment by AndyVojambon** [comment]
This works for me: 
 - write "exit" in the interactive shell, then press enter
 - then select Boot Maintenance Manager > Boot From File > Choose a Volume > EFI > <Your_VM_OS> > select the .efi file
 - press enter and my VM start 🙂

**Reply by scratch-iiit** [reply]
After the Boot From File an empty screen with file explorer shows
Being done for Kali Linux though

**Comment by acertuche368** [comment]
I figured it out while trying to do the same thing:

As long as you selected an iso file then you must catch the option to boot from CD while the VM is starting up. Press any key and it will pull the ISO file and run it or else it will just skip over it and get you to the UEFI Shell.

**Reply by scratch-iiit** [reply]
kept pressing any key but still did not work

**Reply by svnv-svsv-jm** [reply]
Worked for me!

**Reply by wcy18** [reply]
uesfel!!

**Reply by davidlee1229** [reply]
worked..thankyouTT

**Reply by ghost** [reply]
thank you very much worked

**Comment by AnotherWayIn** [comment]
which key is the 'any' key? sorry

**Reply by nicolas17** [reply]
![Where's the any key?](https://media.tenor.com/4D3DhzVOu-oAAAAC/press-any-key-any-key.gif)

**Comment by jsmayo** [comment]
Stumbled upon this after figuring out the solution loool. Just confirming that you need to 'press any key' to initiate the Windows installer. Before starting the instance if you look in configuration portion for the DVD/CD, you'll see the installer pre-loaded 😃 

Also, here's a help link for any future readers: https://docs.getutm.app/guides/windows/

**Comment by rikikamano** [comment]
https://github.com/utmapp/UTM/discussions/3486#discussioncomment-4071577

awesome answer

**Reply by 19KaliNoob87** [reply]
![IMG_20240517_024835330_HDR.jpg](https://github.com/utmapp/UTM/assets/163514066/2ad5724b-edf0-4415-824e-67924c9d7ddc)

I keep getting 2.70 uefi error.

**Comment by dr1np** [comment]
Try press enter repeatedly as fast as you can once you start the VM. I mean really really fast. This solve my problem.

**Comment by 19KaliNoob87** [comment]
> I got same situation.

I second this.

**Comment by iceemo** [comment]
can anyone explain this ti me please?


<img width="791" alt="Screenshot 2567-10-30 at 16 19 30" src="https://github.com/user-attachments/assets/55192276-7464-42ea-97bb-16e80d4174d4">
